propagation.go

```diff
@@ -115,6 +116,9 @@ func NewSplitTextCarrier() *SplitTextCarrier {
 // The Span is separated in this way for a variety of reasons; the most
 // important is to give OpenTracing users a portable way to opt out of Trace
 // Attribute propagation entirely if they deem it a stability risk.
+//
+// Both byte slices may be nil; what is provided will be cleared and the
```

Comment:
> what is provided will be cleared

The carriers are used for both injection and extraction... what does the above mean in the injection case? (Naively it sounds like the contents will be cleared/ignored, though of course that's not what we mean)